PublicMapping/districtbuilder
# Fix sidebar and EqualPopulation metric for population deviation of 0

## Overview

- Fixes the flag in ProjectSidebar to display correctly with a population deviation of 0
- Fixes choropleth steps for Equal Population metric for a population deviation of 0
- Fixes computation of EqualPopulation metric for evaluate mode for a population deviation of 0
- Distinguishes between the unique cases of a whole integer `avgDistrictPopulation`, and an average population that contains a fraction

### Checklist


### Demo

Whole integer avgPopulation

![image](https://user-images.githubusercontent.com/66973361/120707301-27f69580-c488-11eb-8f3d-47c4464d58be.png)

![image](https://user-images.githubusercontent.com/66973361/120707348-35138480-c488-11eb-885d-0985e9f94729.png)



Non-integer avgPopulation

![image](https://user-images.githubusercontent.com/66973361/120709091-6d1bc700-c48a-11eb-8131-973b7016d363.png)

![image](https://user-images.githubusercontent.com/66973361/120709138-7a38b600-c48a-11eb-94f6-bfe6a8110583.png)


## Testing Instructions

- Create a map in PA with an even number of districts (e.g., 4), so the average population contains a fractional remainder
- Add a district that exactly matches the target population, or target population + 1 - expect it to be treated as equal population
- Add a district that is -1 the target population, expect to be treated as invalid

- Create a map in VA with an even number of districts (e.g., 4), so the average population is a whole integer
- Add a district that exactly matches the target population - expect it to be treated as equal population
- Add a district that not exactly equal to the target population, expect to be treated as invalid


Closes #801
